thorlcr/activities/thactivityutil.cpp

```diff
@@ -783,9 +783,42 @@ IFileIO *createMultipleWrite(CActivityBase *activity, IPartDescriptor &partDesc,
     Owned<IFileIO> fileio;
     if (compress)
     {
+        // rowdif used if recordSize > 0, fallback set to LZ4
+        unsigned __int64 compMethod = COMPRESS_METHOD_LZ4;
+        // fallback override
+        StringBuffer compType;
+        activity->getOpt(THOROPT_FBACK_COMPRESS_TYPE, compType);
```

Comment:
This code calls createCompressedFileWriter() which will use the compressor type specified or if the file already exists, it will use the compressor type that was originally used when the file was created.  Could the compatibility issue you mention also happen with the existing code where the option check can change it from row diff to lzw ?